C/Cpp_include/c4Replicator.hh

```diff
@@ -34,15 +34,15 @@ struct C4Replicator
 
     virtual void start(bool reset = false) noexcept = 0;
     virtual void stop() noexcept                    = 0;
-    bool         retry();
+    bool         retry() const;
 
     virtual void stopCallbacks() noexcept = 0;
 
     virtual void setHostReachable(bool) noexcept {}
 
     virtual void setSuspended(bool) noexcept = 0;
 
-    void         setOptions(slice optionsDictFleece);
+    void         setOptions(slice optionsDictFleece) const;
```

Comment:
Weird, why is it constant set method?

Reply:
I would guess because the method doesn't modify anything that's part of C4Replicator, instead it converts `this` to `C4ReplicatorImpl` (which inherits from `C4Replicator`) and modifies a variable of that.

Reply:
Logically, it's a set-method. Why do you want to change it to const method, witch actually would change the underlying replicator?

Reply:
This was just as recommended by Clang-Tidy. To the compiler, this is a const method, because of the weird way we are handling the Replicator / ReplicatorImpl. I agree though, logically it makes more sense of non-const. I can change it.

Reply:
I guess it's because of this, 
static C4ReplicatorImpl* asInternal(const C4Replicator* repl) { return (C4ReplicatorImpl*)repl; }
which turned a const pointer to non-const pointer. Perhaps it should return const and add another overload
static C4ReplicatorImpl* asInternalC4Replicator* repl) { return (C4ReplicatorImpl*)repl; }

Reply:
I don't know, I don't want to change too much as I don't understand that too well. For now, I've reverted the `const` of this method.